zip-0317b.rst

```diff
@@ -0,0 +1,161 @@
+
+Motivation
+==========
+
+As the Zcash blockchain moves towards the Proportional Transfer Fee Mechanism proposed ZIP 317 [#zip-0317]_, transactions that use the ZSA protocol must adapt to this mechanism and include fees consistent with ZIP 317 [#zip-0317]_. 
+
+In this ZIP we propose an adaptation of ZIP 317 [#zip-0317]_ as it applies to ZSA transfer transactions and explore ZSA-specific considerations such as fees for asset issuance.
+
+Specifically, the ZSA transfer and burn mechanism should follow the same fee mechanism in order to not discriminate between transfer bundle types. When it comes to Issuance of ZSA, however, we believe that there should be a preventative incentive that will stop users from flooding the chain with useless asset identifiers.
+
+There are two main factors that will affect the fee mechanism:
+
+- The transaction size, which may take a big part of the block
+- The computational power needed to verify and mine the transaction
+
+In the case of Issuance, the latter is not so relevant as the computational power needed to verify the bundle is not large. The transaction size, however, can be an issue as the number of output notes can be large. Furthermore, as defined in ZIP 227 [#zip-0227]_, there is a new struct in the global state that needs to be maintained as part of the consensus, which motivates further the addition of an Issuance-specific fee.
+
+Requirements
+============
+
+* The conventional fee formula should not favour or discriminate against any
+  of the Orchard, Sapling, or transparent protocols.
+* The fee for a transaction should scale linearly with the number of inputs
+  and/or outputs.
+* Users should be discouraged from creating new “garbage” custom assets
+* Users should not be penalised for sending transactions constructed
+  with padding of inputs and outputs to reduce information leakage.
+  (The default policy employed by zcashd and the mobile SDKs pads to
+  two inputs and two outputs for each shielded pool used by the transaction).
+* Users should be able to spend a small number of UTXOs or notes with value
+  below the marginal fee per input.
+
+
+Specification
+=============
+
+The specification of this ZIP is based on the Proportional Transfer Fee Mechanism ZIP 317 [#zip-0317]_. We expand on how this mechanism will be affected by the replacement of the ZSA protocol for the Orchard protocol.
+
+Issuance Action Fee
+-------------------
+
+As reasoned above, we propose a fee for each issuance action, `issuance_fee`, in order to prevent users from issuing "garbage" assets and to incentivize miners to maintain a more complex blockchain state.
+The proposed fee mechanism is that for each new asset type (i.e.: for each issuance action), the issuer will pay a fixed fee for the action itself, of `0.01 ZEC` or `1,000,000 ZATS`, and a proportional fee for each output note generated for that asset (which is kept in line with the proportional fee mechanism from ZIP 317 [#zip-0317]_).
```

Comment:
```suggestion
The proposed fee mechanism is that for each new asset type (i.e.: for each issuance action), the issuer will pay a fixed fee for the action itself, of `0.01 ZEC` or `1,000,000 ZATS`, and a proportional fee for each output note generated for that asset, including issuance outputs (which is kept in line with the proportional fee mechanism from ZIP 317 [#zip-0317]_).
```

Reply:
This file has been updated and is now tracked in [PR#667](https://github.com/zcash/zips/pull/667).